blog/2023/02/the-injectivity-radii-of-the-unitary-groups.md

```diff
@@ -0,0 +1,206 @@
+$$
+@@important
+    To find the injectivity radii, we only need to find for what values $\theta_i$ is $\exp(\mathrm{i}\theta_i)$ invertible.
+@@
+
+## The injectivity radii
+
+We are finally ready to work out the injectivity radii.
+For $\mathrm{U}(n, \mathbb{F})$, the eigenvalues of $X$ are $\lambda_i = \mathrm{i} \theta_i$ for $\theta_i \in \R$.
+
+The matrix exponential is invertible when $\exp(\mathrm{i} \theta_i)$ is invertible.
+But this is just rotation in the complex plane by the angle $\theta_i$, which is invertible for $\theta_i \in (-\pi, \pi]$.
+The injectivity radii are then computed by the constraint $|\theta_i| = \pi$ for at least one $i \in 1\ldots n$. 
+
+For $\mathrm{U}(n, \mathbb{F})$ and $\mathbb{F} \not\equiv \R$, the largest value of $\norm{\theta}$ achievable subject to the constraints occurs when $|\theta_i| = \pi$ for all $i$.
+Then $\norm{\theta} = \pi \sqrt{n}$.
+The smallest value of $\norm{\theta}$ occurs when $|\theta_i| = \begin{cases}\pi, & i = k\\ 0, & i \ne k\end{cases}$ for any $k \in 1\ldots n$.
+So
+@@important
+$$
+\begin{aligned}
+\operatorname{inj}^-_{\mathrm{U}(n, \mathbb{F})} &= \pi, \qquad \mathbb{F} \not\equiv \R\\
+\operatorname{inj}^+_{\mathrm{U}(n, \mathbb{F})} &= \pi \sqrt{n}.
+\end{aligned}
+$$
+@@
+
+For $\mathrm{SU}(n)$, we have the additional constraint that $\sum_{i=1}^n \theta_i = 0$.
+For even $n$, $\norm{\theta}$ is maximized subject to these constraints when $n/2$ entries in $\theta$ are $+\pi$ and when $n/2$ entries are $-\pi$.
+On the other hand, for odd $n$, it is maximized when $(n-1)/2$ values each are $+\pi$ and $-\pi$ and the $n$th value is $0$. 
+$\norm{\theta}$ is minimized subject to these constraints when there is a single nonzero pair $\theta_i = -\theta_j = \pi$ for $j \ne i$.
+As a result,
+@@important
+$$
+\begin{aligned}
+\operatorname{inj}^-_{\mathrm{SU}(n)} &= \pi\sqrt{2}\\
+\operatorname{inj}^+_{\mathrm{SU}(n)} &= \pi \sqrt{2 \lfloor n/2 \rfloor}.
+\end{aligned}
+$$
+@@
+
+The constraints on $\theta$ required for $\mathrm{O}(n)$ and $\mathrm{SO}(n)$ are also satisfied by the lower and upper bounds of the norms considered for $\mathrm{SU}(n)$, so
+@@important
+$$
+\begin{aligned}
+\operatorname{inj}^-_{\mathrm{O}(n)} = \operatorname{inj}^-_{\mathrm{SO}(n)} &= \pi\sqrt{2}\\
+\operatorname{inj}^+_{\mathrm{O}(n)} = \operatorname{inj}^+_{\mathrm{SO}(n)} &= \pi \sqrt{2 \lfloor n/2 \rfloor}.
+\end{aligned}
+$$
+@@
+
+<!-- It's always a good idea to numerically check that the results make sense.
+For the 2D and 3D rotations $\mathrm{SO}(2)$ and $\mathrm{SO}(3)$, respectively, we then have
+$$\operatorname{inj}^-_{\mathrm{SO}(2)} = \operatorname{inj}^+_{\mathrm{SO}(2)} = \pi\sqrt{2}$$
+$$\operatorname{inj}^-_{\mathrm{SO}(3)} = \operatorname{inj}^+_{\mathrm{SO}(3)} = \pi\sqrt{2}$$
+Coming back to our motivating example, if we start from any pose and rotate an object more than 180° in any direction, we can no longer uniquely determine the initial rotational velocity.
```

Comment:
Here one also sees why Hyper uses the 1/2 in the metric, then you get infectivity radius = pi which is 180 degrees :)

Reply:
But I think this commented out part – besides the note with the sqrt(2) would be nice to have in there as well.

Reply:
I think the 1/2 factor is quite natural for SO(n) for this reason, but not for other number systems (e.g. U(1)). But I like not using the factor early on for simplicity; we have a number of manifolds that inherit the Euclidean inner product from the embedding, and it's nice to not special-case this one manifold.

Reply:
Suer in the end it is a matter of taste, quite a few references I read work with the 1/2 due to skew symmetric matrices and the fact that you get the angle-interpretation directly.